apps/server/src/api-data/db/db.parser.ts

```diff
@@ -20,7 +21,21 @@ type ParsingError = {
  * @param {object} jsonData - project file to be parsed
  * @returns {object} - parsed object
  */
-export function parseDatabaseModel(jsonData: Partial<DatabaseModel>): { data: DatabaseModel; errors: ParsingError[] } {
+export function parseDatabaseModel(jsonData: Partial<DatabaseModel>): {
+  data: DatabaseModel;
+  errors: ParsingError[];
+  migrated: boolean;
+} {
+
+  //TODO: TEST THIS!!!!!!!
+  let migrated = false;
+  if (v3.shouldUseThisMigration(jsonData)) {
+    migrated = true;
+    logger.warning(LogOrigin.Server, 'The imported project is from v3, trying to migrate');
+    //TODO: is this bad?
+    jsonData = v3.migrateAllData(jsonData);
```

Comment:
I would personally prefer it if we created a copy of the migrated data
there is also a fair share of parsing going on, so we may want to have this inside a try-catch